pkg/shp/reactor/pod_watcher.go

```diff
@@ -0,0 +1,127 @@
+package reactor
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+// PodWatcher a simple function orchestrator based on watching a given pod and reacting upon the
+// state modifications, should work as a helper to build business logic based on the build POD
+// changes.
+type PodWatcher struct {
+	ctx     context.Context
+	stopCh  chan bool       // stops the event loop execution
+	watcher watch.Interface // client watch instance
+
+	skipPodFn       SkipPodFn
+	onPodAddedFn    OnPodEventFn
+	onPodModifiedFn OnPodEventFn
+	onPodDeletedFn  OnPodEventFn
+}
+
+// SkipPodFn a given pod instance is informed and expects a boolean as return. When true is returned
+// this container state processing is skipped completely.
+type SkipPodFn func(pod *corev1.Pod) bool
+
+// OnPodEventFn when a pod is modified this method handles the event.
+type OnPodEventFn func(pod *corev1.Pod) error
```

Comment:
returning the watched object applies here too

Reply:
Please consider my previous comment ([here](https://github.com/gabemontero/cli-1/pull/1#discussion_r655865342)).